test/handlers/FranchiserFactoryHandler.sol

```diff
@@ -15,76 +15,234 @@ contract FranchiserFactoryHandler is Test {
 
-    // Handler ghost array to contain all the funded franchisers created by the last call to handler_fundMany
+    // Handler ghost AddressSet to contain all of the delegators that were used to fund franchisers
+    EnumerableSet.AddressSet private delegators;
+
+    // Handler ghost AddressSet to contain all of the delegatees that were delegated to by handler functions
+    EnumerableSet.AddressSet private delegatees;
+
+    // Handler ghost AddressSet to contain all of the Franchisers created by sub-delegation
+    EnumerableSet.AddressSet private subDelegatedFranchisers;
+
+    // Handler ghost array to contain all the funded franchisers created by the last call to factory_fundMany
     Franchiser[] private lastFundedFranchisersArray;
 
-    // Ghost variable address to receive of the total amount of funds recalled from franchisers
-    address public targetAddressForRecalledFunds = makeAddr("targetAddressForRecalledFunds");
+    // Handler ghost array to contain all the franchisers created by the last call to franchiser_subDelegateMany
+    Franchiser[] private lastSubDelegatedFranchisersArray;
+
+    // Franchiser that performed the last sub-delegation
+    Franchiser private lastSubDelegatingFranchiser;
 
     constructor(FranchiserFactory _factory) {
         factory = _factory;
         votingToken = VotingTokenConcrete(address(factory.votingToken()));
         franchiser = new Franchiser(IVotingToken(address(votingToken)));
+        balances_updated_correctly = true;
+        voting_powers_updated_correctly = true;
     }
 
     modifier countCall(bytes32 key) {
         calls[key].calls++;
         _;
     }
 
-    function sumFundedFranchisersBalances() public view returns (uint256 sum) {
-        for (uint256 i = 0; i < fundedFranchisers.length(); i++) {
-            sum += factory.votingToken().balanceOf(fundedFranchisers.at(i));
-        }
-    }
-
     function _validActorAddress(address _address) internal view returns (bool valid) {
         valid = (_address != address(0))
             && (
                 _address != address(votingToken) && (_address != address(factory))
-                    && (!fundedFranchisers.contains(_address))
+                    && (!fundedFranchisers.contains(_address)
+                    && (!subDelegatedFranchisers.contains(_address)))
             );
     }
 
     function _boundAmount(uint256 _amount) internal pure returns (uint256) {
         return bound(_amount, 0, 100_000_000e18);
     }
 
-    function handler_fund(address _delegator, address _delegatee, uint256 _amount) external countCall("handler_fund") {
+    // function to get the amount of voting power held by a franchiser for a given delegator and delegatee
+    // (used to get balance of a franchiser via pre-calculated address that may have been deployed previously)
+    function _getAmountInFranchiserGivenDelegatorAndDelegatee(address _delegator, address _delegatee) internal view returns (uint256) {
```

Comment:
Renamed with slightly different name since it also now gets balance from sub-delegation tree: https://github.com/uniswapfoundation/franchiser/blob/bfbb0b20a33886dd1a6753e5a45727bef6aa637a/test/handlers/FranchiserFactoryHandler.sol#L81